src/backends/gr.jl

```diff
@@ -1188,49 +1217,44 @@ function gr_update_viewport_legend!(vp, sp, leg)
     xaxis, yaxis = sp[:xaxis], sp[:yaxis]
     xmirror = mirrored(xaxis, :top)
     ymirror = mirrored(yaxis, :right)
-    if (lp = sp[:legend_position]) isa Tuple{<:Real,Symbol}
-        if lp[2] === :outer
-            x, y = gr_legend_pos(sp, leg, vp) # Dry run, to figure out
-            if x < vp.xmin
-                vp.xmin +=
-                    leg.leftw +
-                    leg.textw +
-                    leg.rightw +
-                    leg.xoffset +
-                    !ymirror * gr_axis_width(sp, yaxis)
-            elseif x > vp.xmax
-                vp.xmax -= leg.leftw + leg.textw + leg.rightw + leg.xoffset
-            end
-            if y < vp.ymin
-                vp.ymin +=
-                    leg.h + leg.dy + leg.yoffset + !xmirror * gr_axis_height(sp, xaxis)
-            elseif y > vp.ymax
-                vp.ymax -= leg.h + leg.dy + leg.yoffset
-            end
-        end
+    leg_str = if (lp = sp[:legend_position]) isa Tuple{<:Real,Symbol} && lp[2] === :outer
+        x, y = gr_legend_pos(sp, leg, vp)  # dry run, to figure out
+        horz = x < vp.xmin ? "left" : (x > vp.xmax ? "right" : "")
+        vert = y < vp.ymin ? "bot" : (y > vp.ymax ? "top" : "")
+        "outer" * vert * horz
+    else
+        string(lp)
     end
-    leg_str = string(lp)
     if occursin("outer", leg_str)
+        xoff = if leg.vertical
+            leg.pad + leg.span + leg.textw + leg.pad + leg.xoffset
+        else
+            leg.w + leg.dx + leg.xoffset
+        end
+        yoff = if leg.vertical
+            leg.h + leg.dy + leg.yoffset
+        else
+            leg.pad + leg.span + leg.texth + leg.pad + leg.yoffset
+        end
         if occursin("right", leg_str)
-            vp.xmax -= leg.leftw + leg.textw + leg.rightw + leg.xoffset
+            vp.xmax -= xoff
         elseif occursin("left", leg_str)
-            vp.xmin +=
-                leg.leftw +
-                leg.textw +
-                leg.rightw +
-                leg.xoffset +
-                !ymirror * gr_axis_width(sp, yaxis)
+            vp.xmin += xoff + !ymirror * gr_axis_width(sp, yaxis)
         elseif occursin("top", leg_str)
-            vp.ymax -= leg.h + leg.dy + leg.yoffset
-        elseif occursin("bottom", leg_str)
-            vp.ymin += leg.h + leg.dy + leg.yoffset + !xmirror * gr_axis_height(sp, xaxis)
+            vp.ymax -= yoff
+        elseif occursin("bot", leg_str)  # NOTE: matches `bottom` or `bot`
+            vp.ymin += yoff + !xmirror * gr_axis_height(sp, xaxis)
         end
     end
     if lp === :inline
-        if yaxis[:mirror]
-            vp.xmin += leg.w
+        if leg.vertical
+            if yaxis[:mirror]
+                vp.xmin += leg.w
+            else
+                vp.xmax -= leg.w
+            end
         else
-            vp.xmax -= leg.w
+            # FIXME: update these when vertical
```

Comment:
what should happen here? (and you probably meant horizontal)

Reply:
Thanks, do you know what `:inline` does ?

Reply:
I don't really know, but I would expect it to show the labels at the end of the lines.

Reply:
probably this: https://github.com/JuliaPlots/Plots.jl/pull/2521

Reply:
I'd say you can leave it as it was before

Reply:
Done, I've checked with `plot(rand(10, 4), legend = :inline, xflip = true)`, this was not concerned by the layout of the legend box.

Reply:
What about `plot(rand(10, 4), legend = :inline, ymirror = true, legend_columns = -1)` ?

Reply:
![inline](https://user-images.githubusercontent.com/13423344/200165070-c07a7de4-c10e-4842-8740-625c166a6848.png)

Left padding is wrong, I will fix this.